### PR TITLE
Sort tasks by priority

### LINKED ISSUE #69: Sort Tasks by priority
As a student, I want to list my tasks sorted by priority so that I can know what tasks to work on first.

### PR DESCRIPTION
**Pull request by GuoAi**
Fixes #69

### CODE REVIEW

File: src/main/java/seedu/duke/commands/ListCommand.java [hunk 1]
```diff
@@ -78,9 +79,8 @@ public void execute(TaskList tasks) throws DukeException {
             TaskList newTaskList = new TaskList(newTasks);
             newTaskList.listTask(category);
         } else {
-            for (int i = 0; i < tasks.size(); i++) {
-                newTasks.add(tasks.get(i));
-            }
+            newTasks = tasks.getTaskList();
+            Collections.sort(newTasks);
             TaskList newTaskList = new TaskList(newTasks);
             newTaskList.listTask();
```
Comment (iamchenjiajun) [inline src/main/java/seedu/duke/commands/ListCommand.java:97]: I was thinking that if we only print a new sorted list using the "list" command, the users will never know which task they are setting as "done" or "deleted".

Comment (GuoAi) [reply src/main/java/seedu/duke/commands/ListCommand.java:97]: Ya. Maybe we can have this command as "list tasks sorted" while keeping the original "list" command as unchanged?

Comment (iamchenjiajun) [reply src/main/java/seedu/duke/commands/ListCommand.java:97]: Yea that makes sense.

**Review by iamchenjiajun — APPROVED** [review]
LGTM!

**Comment by MuhammadHoze** [comment]
LGTM!

**Comment by iamchenjiajun** [comment]
There's so many constructors to ListCommand now, I wonder if we can actually pass in an enum to the ListCommand instead and use a switch statement in the execute method. But that will be for another time I guess (maybe v2.1 or another PR sometime later).

But then with enums we'd have to pass in blank arguments for "category" when listing category, so another point to consider.

**Comment by MuhammadHoze** [comment]
yeah true, there will be a few more coming in soon.  yeap sure 2.1

**Comment by GuoAi** [comment]
Or maybe we can just have one big constructor ListCommand, i.e. ListCommand(int priority, String category, boolean isSorted) and pass these arguments in CommandCreator?

**Pull request merged**